src/MAUI/Maui.Samples/Samples/Location/LocationWithNMEA/LocationWithNMEA.xaml.cs

```diff
@@ -147,92 +140,95 @@ public void Dispose()
      * For NMEA location data sources created from a Bluetooth device or serial input, you may not need to create your own stream.
-     * For any other case, you can write the data to a memory stream like below.
+     * For any other case, you can simulate the datastream from a file like below
      */
 
-    public class NMEAStreamSimulator : IDisposable
+    public class SimulatedNmeaStream : Stream
     {
-        private Timer _timer;
+        private const int DefaultInterval = 1000; // he default interval in milliseconds between bursts of NMEA data.
 
-        public Stream MessageStream;
+        private readonly Timer _timer;
 
-        private int _lineCounter = 0;
-        private const int DefaultInterval = 1000;
+        private readonly StreamReader _sr;
 
-        private string[] _nmeaStrings;
+        private int _pendingBursts = 0;
 
-        public event EventHandler<NmeaMessageEventArgs> NmeaMessageChanged;
-
-        public NMEAStreamSimulator(double speed = 1.0)
+        public SimulatedNmeaStream(string file)
         {
-            _timer = new Timer(DefaultInterval / speed);
-
             // Populate an array with all of the mock NMEA data.
-            string nmeaMockDataPath = DataManager.GetDataFolder("d5bad9f4fee9483791e405880fb466da", "Redlands.nmea");
-            _nmeaStrings = File.ReadAllText(nmeaMockDataPath).Split('\n');
-
-            // Create a data stream for the `NmeaLocationDataSource` to use.
-            MessageStream = new MemoryStream();
-
+            _sr = new StreamReader(file);
+            _timer = new Timer(DefaultInterval);
             _timer.Elapsed += TimerElapsed;
-        }
-
-        public void Start()
-        {
             _timer.Start();
         }
 
-        public void Stop()
-        {
-            _timer.Stop();
-        }
+        private void TimerElapsed(object sender, ElapsedEventArgs e) => Interlocked.Increment(ref _pendingBursts);
 
-        public void Reset()
+        protected override void Dispose(bool disposing)
         {
-            _timer.Stop();
-            _lineCounter = 0;
+            if (disposing)
+            {
+                // Dispose of the timer and stream.
+                _timer.Stop();
+                _timer.Dispose();
+                _sr.Dispose();
+            }
+            base.Dispose(disposing);
         }
 
-        private void TimerElapsed(object sender, ElapsedEventArgs e)
-        {
-            // Get the next NMEA string and append return and newline characters to it.
-            string nmeaString = $"{_nmeaStrings[_lineCounter]}\r\n";
+        #region Implementation of a custom System.IO.Stream 
 
-            NmeaMessageChanged?.Invoke(this, new NmeaMessageEventArgs { NmeaMessage = nmeaString });
+        public override int Read(byte[] buffer, int offset, int count)
+        {
+            if (_pendingBursts == 0)
+                return 0; //Nothing in the buffer to read
```

Comment:
Would you mind applying the commenting style used in this repo to comments in this sample?
- no in-line comments
- sentence should start with capitalization and end with a full stop
- should be a space between `//` and first letter of comment